### ISSUE TITLE
No dependency to extension recorded for external protocols.

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#2942]
If you create an extension that includes an external protocol, no dependency is recorded between the external protocol and the extension. That means that if you create an extension that installs an external protocol, and then issue DROP EXTENSION, it will fail.

As part of the extensions patch in PostgreSQL, all the functions that handle CREATE commands were modified to record the dependency, but the GPDB-specific CREATE PROTOCOL command was missed. For example, the patch did this to pg_conversion.c:
```
--- a/src/backend/catalog/pg_conversion.c
+++ b/src/backend/catalog/pg_conversion.c
@@ -128,6 +128,8 @@ ConversionCreate(const char *conname, Oid connamespace,
        /* create dependency on owner */
        recordDependencyOnOwner(ConversionRelationId, HeapTupleGetOid(tup),
                                                        conowner);
+       /* dependency on extension */
+       recordDependencyOnCurrentExtension(&myself, false);
 
        heap_freetuple(tup);
        heap_close(rel, RowExclusiveLock);
```

A `recordDependencyOnCurrentExtension` call like that also needs to be added to ExtProtocolCreate. So this isn't a big change. In order to test this, I think we should turn the `contrib/extprotocol` demo into an extension, and also include DROP EXTENSION in its regression tests.

**Comment by hlinnaka** [comment]
Now that the PXF extension is in, which is an external protocol packaged in an extension, this is easy to repdroduce:
```
postgres=# create extension pxf;
CREATE EXTENSION
postgres=# drop extension pxf;
NOTICE:  protocol pxf depends on function pxf_read()
ERROR:  cannot drop extension pxf because other objects depend on it
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
```

**Comment by hlinnaka** [comment]
Fixed by PR #3087.
